bitwarden_license/src/Commercial.Core/SecretsManager/Commands/AccessTokens/CreateAccessTokenCommand.cs

```diff
@@ -55,7 +57,22 @@ public async Task<ApiKey> CreateAsync(ApiKey apiKey, Guid userId)
             throw new NotFoundException();
         }
 
-        apiKey.ClientSecret = CoreHelpers.SecureRandomString(_clientSecretMaxLength);
-        return await _apiKeyRepository.CreateAsync(apiKey);
+        var clientSecret = CoreHelpers.SecureRandomString(_clientSecretMaxLength);
+
+        // Store the hash of the client secret in the database.
+        apiKey.HashedClientSecret = GetHash(clientSecret);
+        var result = await _apiKeyRepository.CreateAsync(apiKey);
+
+        // Return the plain text client secret so the client can generate the access token.
+        result.HashedClientSecret = clientSecret;
```

Comment:
This seems somewhat sub-optimal since you could accidentally save it at a later stage. I think it's better to define a DTO that contains apiKey and plain text key, that is used as an input to the response model.

Reply:
Resolved in https://github.com/bitwarden/server/pull/2943/commits/3c640e6fe71ce8f18864693f0aa63bdf8ab57cdb